src/linear_algebra/basic.lean

```diff
@@ -758,16 +758,23 @@ preserved under addition and scalar multiplication, then `p` holds for all eleme
   (H2 : ∀ (a:R) x, p x → p (a • x)) : p x :=
 (@span_le _ _ _ _ _ _ ⟨p, H0, H1, H2⟩).2 Hs h
 
-lemma span_eq_add_submonoid.closure {M : Type*} [add_comm_monoid M] (S : set M) :
-  (span ℕ S).to_add_submonoid = add_submonoid.closure S :=
+section span_nat
+open submodule
+
+lemma span_nat_eq_add_group_closure (s : set R) :
```

Comment:
```suggestion
lemma span_nat_eq_add_submonoid_closure (s : set R) :
```
I also suspect `R` probably isn't the right type, as it's likely to be a `semiring`

Reply:
Ah, sure!